packages/legend-graph/src/graphManager/action/EngineWarning.ts

```diff
@@ -0,0 +1,28 @@
+/**
+ * Copyright (c) 2020-present, Goldman Sachs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { ApplicationError } from '@finos/legend-shared';
+import type { SourceInformation } from './SourceInformation.js';
+
+export class EngineWarning extends ApplicationError {
```

Comment:
hm not really i originally had it as an own class but was thinking maybe it should count as an error technically with the future plans. but will switch to not extend since true it is called `EngineWarning` 👍